python/cudf/cudf/core/series.py

```diff
@@ -2332,6 +2338,40 @@ def apply(self, func, convert_dtype=True, args=(), **kwargs):
         1    <NA>
         2     4.5
         dtype: float64
+
+        String data is allowed, so long as its use does not modify or create
+        a new string. The following UDF is allowed:
+
+        >>> def f(st):
+        ...     if len(st) == 0:
+        ...             return -1
+        ...     elif st.startswith('a'):
+        ...             return 1
+        ...     elif 'example' in st:
+        ...             return 2
+        ...     else:
+        ...             return 3
+        ...
+        >>> sr = cudf.Series(['', 'abc', 'some_example'])
+        >>> sr.apply(f)
+        0   -1
+        1    1
+        2    2
+        dtype: int64
+
+        However the following UDF is not allowed, because it includes an
+        operation that requires the creation of a new string, namely the
+        ``upper`` method. Methods that are not supported in this manner
+        will raise an ``AttributeError``.
+        >>> def f(st):
+        ...     new = st.upper()
+        ...     return 'ABC' in new
+        ...
+        >>> sr.apply(f) # AttributeError
+
+        For a complete list of supported functions and methods that may be used
+        to manipulate string data, see the ``guide-to-udfs`` IPython notebook.
```

Comment:
If you like the suggestions above, apply them mutatis mutandis to this docstring as well.